home/views/api/leaderboard.py

```diff
@@ -0,0 +1,365 @@
+        try:
+            device = Device.objects.get(device_id=json_data["account_id"])
+        except ObjectDoesNotExist:
+            return JsonResponse(
+                {
+                    "status": "error",
+                    "message": (
+                        "Unregistered device - "
+                        f"{json_data['account_id']}."
+                        " Please register first!"
+                    ),
+                }
+            )
+        # Json response template
+        json_response = {
+            "status": "success",
+            "message": "Leaderboard accessed successfully",
+            "payload": {
+                "account_id": device.device_id,
+                "leaderboard": [],
+            },
+        }
+
+        # def get(self, request, *args, **kwargs):
+        # user_stats_container = {}
+
+        contest = Contest.active()
+
+        # daily_walks, intentional_walks, _, _ = get_contest_walks(contest)
+
+        # # get the current/next Contest
+        leaderboard_list = []
+        # user_steps = {}
+        # for email, dw_row in daily_walks.items():
+        #     acct = Account.objects.values(*ACCOUNT_FIELDS).get(email=email)
+        #    # print("acct", acct)
+        #     accountid = acct.get("id")
+        #     dev_id = (
+        #         Account.objects.values("device")
+        #         .filter(id=accountid)
+        #         .reverse()[0]
+        #     )
+        #     #print("dev_id", dev_id)
+        #     dev = Device.objects.values(*DEVICE_FIELDS).filter(
+        #         account_id=accountid
+        #     )[0]
+        #    # print("device id", dev)
+        #    # print("iso d", dev["device_id"])
+        #     # print("id, ",Device.objects.get("device_id"))
+        #     if contest and acct["created"] > localize(contest.end):
+        #         continue
+
+        #     user_stats = user_stats_container.get(
+        #         email, dict(new_signup=False)
+        #     )
+        #     user_stats["account"] = acct
+        #     user_stats["dw_steps"] = dw_row["dw_steps"]
+        #     id = acct["id"]
+        #     # device = Device.objects.get(device_id = acct)
+        #     # print(device)
+
+        #     # id = device.device_id
+        #     # device = Device.objects.get(device_id=json_data["account_id"])
+        #     rank = -1
+        #     user_steps[id] = dw_row["dw_steps"]
+        #     leaderboard_list.append(
+        #         {"rank":rank, "id": dev["device_id"], "steps": user_steps[id]}
+        #     )
+        contest = Contest.active()
+        leaderboard_list = []
+        leaderboard_length = 10
+
+        # c = [(idx, item) for idx,item in enumerate(leaderboard_list, start=1)]
+
+        def build_dict(seq, key):
+            return list(
+                (dict(d, rank=rank + 1)) for (rank, d) in enumerate(seq)
+            )
+
+        leaderboard = Leaderboard.objects.all()
+        model_leaderboard_list = []
```

Comment:
Instead of all this code to process ALL leaderboard objects, we should be using Django QuerySet capabilities to use the database to do this work for us.

https://docs.djangoproject.com/en/4.1/ref/models/querysets/

First, shouldn't we be doing a `filter` by contest_id?